doc/redeemer.md

```diff
@@ -0,0 +1,58 @@
+# Ticket Redemption Service
+
+The Ticket Redemption Service allows Orchestrator nodes to redeem winning tickets on-chain using a separate ETH account on the Ticket Redemption Service. Orchestrator nodes no longer need to have an account funded with ETH to pay for transactions on the Ethereum network.
+
+It is responsible for redeeming winning tickets as well as pushing _max float_ updates about broadcasters back to its connected Orchestrators.
+
+_Max float_ is the guaranteed value an Orchestrator will be able to claim from a Broadcaster's reserve. It accounts for the current reserve allocation from a Broadcaster to an Orchestrator as well as pending winning ticket redemptions. 
+
+## TicketQueue
+
+1. The `ticketQueue` is a loop that runs everytime a new block is seen. It will then pop tickets off the queue starting with the oldest ticket first, and sends it to the `LocalSenderMonitor` for redemption if the `recipientRand` for the ticket has expired. 
+
+2. When the `LocalSenderMonitor` receives a ticket from the `ticketQueue` it will substract `ticket.faceValue` from the outstanding `maxFloat` as long as the ticket is in limbo. 
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _This will trigger a `LocalSenderMonitor.SubscribeMaxFloatChange(ticket.sender)` notification_
+
+3. The ticket is sent to a remote Ethereum node for redemption
+
+4. When the Ethereum node returns a response the `ticket.faceValue` is added to the `maxFloat` again as the ticket is no longer in limbo.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _This will trigger a `LocalSenderMonitor.SubscribeMaxFloatChange(ticket.sender)` notification_
+
+## Monitoring Max Float
+
+*A more detailed description about `maxFloat` and it's relation to a broadcaster's reserve can be found in the [PM protocol spec](https://github.com/livepeer/wiki/blob/master/spec/streamflow/pm.md#reserve).*
+
+1. When max float for a `sender` is requested from the `RedeemerClient` but no local cache is available, an (unary) RPC call will be sent to the `Redeemer`. 
+
+2. A second RPC call to `MonitorMaxFloat(sender)` will open up a server-side gRPC stream to receive future update. 
+
+_If this call fails the response from step 1 is returned, but not kept in cache to prevent duplicate streams_
```

Comment:
Is this referring to step 2? If so, why would keeping the result from step 1 in the cache result in duplicate streams? If step 2 fails, wouldn't there be no stream at all?

Reply:
Ah yes that's true, this must be an old line that I repurposed. 

I think it should state something like "to prevent stale cache"

Reply:
fixed in fcb86dd293a40d0a61916d875394afeb32a80dd5